### ISSUE TITLE
Setting up root CA cert

### ISSUE THREAD

**Issue by shelane** [docksal/docksal#518]
**Description**

Problem/feature description...
I am trying to install root CA certificates in the CLI for allowing firewall/network SSL inspection according to my work network requirements.

**Steps to reproduce the issue:**

1. Downloaded crt file from network team.
2. Ran commands to create PEM file and saved both in .docksal/openssl
(openssl x509 -text -inform PEM -in cspca.pem >> cert.pem)
3. Copied the files to the directory where certs are setup in cli:
cd /etc/ssl/certs
sudo cp /var/www/.docksal/openssl/cspca.pem cspca.pem
sudo cp /var/www/.docksal/openssl/cert.pem cert.pem

4. Ran instructed test:
openssl s_client -connect google.com:443

**Describe the results you received:**

Results...
gethostbyname failure
connect:errno=11

**Describe the results you expected:**

```
CONNECTED(00000005)
depth=1 C = US, O = Google Trust Services, CN = Google Internet Authority G3
verify error:num=20:unable to get local issuer certificate
verify return:0
---
Certificate chain
 0 s:/C=US/ST=California/L=Mountain View/O=Google Inc/CN=*.google.com
   i:/C=US/O=Google Trust Services/CN=Google Internet Authority G3
 1 s:/C=US/O=Google Trust Services/CN=Google Internet Authority G3
   i:/OU=GlobalSign Root CA - R2/O=GlobalSign/CN=GlobalSign
---
Server certificate
-----BEGIN CERTIFICATE-----
... (truncated)
```

Expected behavior...

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
COMPOSE_PROJECT_NAME_SAFE: drupal8
COMPOSE_FILE:
/Volumes/UserDrive/shelane/.docksal/stacks/overrides-osxfs.yml
/Volumes/UserDrive/shelane/.docksal/stacks/volumes-bind.yml
/Volumes/UserDrive/shelane/.docksal/stacks/stack-acquia.yml
/Volumes/UserDrive/shelane/Sites/docksal/drupal8/.docksal/docksal.yml
ENV_FILE:
/Volumes/UserDrive/shelane/Sites/docksal/drupal8/.docksal/docksal.env

PROJECT_ROOT: /Volumes/UserDrive/shelane/Sites/docksal/drupal8
DOCROOT: docroot
VIRTUAL_HOST: drupal8.docksal
VIRTUAL_HOST_ALIASES: *.drupal8.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32769

Docker Compose configuration
---------------------
services:
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      DOCROOT: docroot
      HOST_GID: '20'
      HOST_UID: '503'
      SECRET_ACAPI_EMAIL: '[*****]'
      SECRET_ACAPI_KEY: '[*****]'
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      SOURCE_ALIAS: null
      VIRTUAL_HOST: drupal8.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: lpeabody/blt-cli
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - /home/docker
    - /Volumes/UserDrive/shelane/Sites/docksal/drupal8:/var/www:rw,cached
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    - 192.168.64.100
    - 8.8.8.8
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_DATABASE: default
      MYSQL_PASSWORD: user
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.1-mysql-5.6
    ports:
    - 3306/tcp
    volumes:
    - project_root:/var/www:ro,nocopy
  memcached:
    command:
    - -m
    - '128'
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: memcached
    image: memcached:1.4-alpine
  solr:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: solr
    image: docksal/solr:1.0-solr4
    labels:
      io.docksal.virtual-host: solr.drupal8.docksal
      io.docksal.virtual-port: '8983'
    volumes:
    - project_root:/var/www:ro,nocopy
  varnish:
    depends_on:
      web:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      VARNISH_BACKEND_HOST: web
    hostname: varnish
    image: docksal/varnish:1.1-varnish5
    labels:
      io.docksal.virtual-host: varnish.drupal8.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    - 192.168.64.100
    - 8.8.8.8
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.project-root: /Volumes/UserDrive/shelane/Sites/docksal/drupal8
      io.docksal.virtual-host: drupal8.docksal,*.drupal8.docksal
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Volumes/UserDrive/shelane/Sites/docksal/drupal8
      o: bind
      type: none

---------------------
```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
Mac OS X 10.13.4
Darwin USSEnterprise 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
Mode: Docker for Mac

███  FIN
fin version: 1.52.0

███  ENVIRONMENT
DOCKER_HOST   | 
 WARNING:  Non-default DOCKER_HOST
------------------------------------------------------------
DOCKSAL_NFS_PATH  | 

███  DOCKER
EXPECTED VERSION: 17.09.0-ce

Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      18.03.1-ce
 API version:  1.37 (minimum version 1.12)
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:22:38 2018
 OS/Arch:      linux/amd64
 Experimental: true

███  DOCKER INFO
Containers: 19
 Running: 13
 Paused: 0
 Stopped: 6
Images: 21
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.87-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 9
Total Memory: 2.933GiB
Name: linuxkit-025000000001
ID: IJ7Z:APUE:JXX4:KOQJ:OS6W:42WA:KNBW:QMH5:ZWQI:FH22:VBIY:ETD6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 132
 Goroutines: 139
 System Time: 2018-05-03T23:20:31.330230113Z
 EventsListeners: 4
Http Proxy: docker.for.mac.http.internal:3128
Https Proxy: docker.for.mac.http.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

███  DOCKER COMPOSE
EXPECTED VERSION: 1.19.0
docker-compose version 1.19.0, build 9e633ef
docker-py version: 2.7.0
CPython version: 2.7.10
OpenSSL version: OpenSSL 0.9.8zh 14 Jan 2016

███  DOCKER MACHINE
EXPECTED VERSION: 0.14.0
docker-machine version 0.14.0, build 89b8332

NAME      ACTIVE   DRIVER       STATE     URL   SWARM   DOCKER    ERRORS
default   -        virtualbox   Stopped                 Unknown   
docksal   -        virtualbox   Stopped                 Unknown   

███  DOCKER: IMAGES
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
nodejs/cli            latest              bce417281c2e        2 weeks ago         885MB
docksal/db            1.2-mysql-5.6       0f0acba4c6d1        3 weeks ago         256MB
docksal/cli           2.1-php7.1          4c7331d4cf77        4 weeks ago         950MB
docksal/cli           latest              b1ca4fa0852d        4 weeks ago         876MB
docksal/cli           2.0-php7.1          9ba7c35bb83f        6 weeks ago         914MB
docksal/cli           2.0-php5.6          0fea463bfbd9        6 weeks ago         887MB
docksal/vhost-proxy   1.2                 a1c980537985        7 weeks ago         166MB
lpeabody/blt-cli      latest              ea0294b1a4ba        8 weeks ago         1.01GB
drupal                latest              59d778ddb4e9        8 weeks ago         452MB
docksal/varnish       1.1-varnish5        967b76b88b85        2 months ago        107MB
cweagans/bg-sync      latest              ebeae43e7c96        3 months ago        16.8MB
hello-world           latest              f2a91732366c        5 months ago        1.85kB
docksal/cli           1.3-php7            19bdddfeab89        7 months ago        759MB
docksal/web           2.1-apache2.4       32c1d32eba6d        8 months ago        87.5MB
docksal/vhost-proxy   1.1                 898fb21e7f2d        9 months ago        158MB
docksal/db            1.1-mysql-5.6       6feb79af9597        9 months ago        298MB
memcached             1.4-alpine          273374b463b8        9 months ago        7.68MB
docksal/dns           1.0                 20c7a535479d        15 months ago       61.6MB
docksal/ssh-agent     1.0                 148220cc033d        15 months ago       12.2MB
docksal/solr          1.0-solr4           fa6b236caf06        16 months ago       546MB
docksal/varnish       1.0-varnish4        35dae6b91ff9        16 months ago       99.5MB

███  DOCKER: CONTAINERS
CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS                       PORTS                                                    NAMES
1a39dc91f02f        docksal/varnish:1.1-varnish5   "/opt/startup.sh"        About an hour ago   Up About an hour             80/tcp, 6082/tcp                                         drupal8_varnish_1
a86680e931a2        docksal/web:2.1-apache2.4      "httpd-foreground"       About an hour ago   Up About an hour             80/tcp, 443/tcp                                          drupal8_web_1
a201b7438023        lpeabody/blt-cli               "/opt/startup.sh s..."   About an hour ago   Up About an hour (healthy)   22/tcp, 3000/tcp, 9000/tcp                               drupal8_cli_1
f9dab876f2e3        docksal/db:1.1-mysql-5.6       "/entrypoint.sh my..."   About an hour ago   Up About an hour             0.0.0.0:32769->3306/tcp                                  drupal8_db_1
76c7b56823d7        docksal/solr:1.0-solr4         "/opt/solr/bin/sol..."   About an hour ago   Up About an hour             8983/tcp                                                 drupal8_solr_1
ecd6a13691a6        memcached:1.4-alpine           "docker-entrypoint..."   About an hour ago   Up About an hour             11211/tcp                                                drupal8_memcached_1
8bd93c3ee44c        docksal/web:2.1-apache2.4      "httpd-foreground"       9 hours ago         Up 9 hours                   80/tcp, 443/tcp                                          llnl_web_1
1b1f4bf7a94f        docksal/cli:2.1-php7.1         "/opt/startup.sh s..."   9 hours ago         Up 9 hours (healthy)         22/tcp, 3000/tcp, 9000/tcp                               llnl_cli_1
e2100b3c6c62        docksal/ssh-agent:1.0          "/run.sh ssh-agent"      7 days ago          Up 9 hours                                                                            docksal-ssh-agent
459b1422a299        docksal/dns:1.0                "/opt/entrypoint.s..."   7 days ago          Up 9 hours                   192.168.64.100:53->53/udp                                docksal-dns
3553b26766f4        docksal/vhost-proxy:1.2        "docker-entrypoint..."   7 days ago          Up 9 hours                   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
f58d1b1d0e57        docksal/db:1.2-mysql-5.6       "/entrypoint.sh my..."   9 days ago          Up 9 hours                   0.0.0.0:32768->3306/tcp                                  llnl_db_1
876323897c5e        cweagans/bg-sync               "bg-sync"                9 days ago          Up 8 hours                                                                            llnl_bg-sync_1

███  VIRTUALBOX
EXPECTED VERSION: 5.1.28
5.1.28r117968

███  VIRTUALBOX NETWORK INTERFACES
Name:            vboxnet0
GUID:            786f6276-656e-4074-8000-0a0027000000
DHCP:            Disabled
IPAddress:       192.168.64.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:00
MediumType:      Ethernet
Status:          Down
VBoxNetworkName: HostInterfaceNetworking-vboxnet0

Name:            vboxnet1
GUID:            786f6276-656e-4174-8000-0a0027000001
DHCP:            Disabled
IPAddress:       192.168.99.1
NetworkMask:     255.255.255.0
IPV6Address:     
IPV6NetworkMaskPrefixLength: 0
HardwareAddress: 0a:00:27:00:00:01
MediumType:      Ethernet
Status:          Down
VBoxNetworkName: HostInterfaceNetworking-vboxnet1

███  NFS EXPORTS

# <ds-nfs docksal
/Users 127.0.0.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>

███  HDD Usage
Filesystem      Size   Used  Avail Capacity  iused               ifree %iused  Mounted on
/dev/disk4s1   238Gi   68Gi  167Gi    29%  1415638 9223372036853360169    0%   /
devfs          198Ki  198Ki    0Bi   100%      684                   0  100%   /dev
/dev/disk4s4   238Gi  3.0Gi  167Gi     2%        3 9223372036854775804    0%   /private/var/vm
/dev/disk1s2   931Gi  420Gi  511Gi    46%    10493          4294956786    0%   /Volumes/UserExtras
map -hosts       0Bi    0Bi    0Bi   100%        0                   0  100%   /net
map auto_home    0Bi    0Bi    0Bi   100%        0                   0  100%   /home
/dev/disk2s2   931Gi  522Gi  409Gi    57%  1357978          4293609301    0%   /Volumes/UserDrive
/dev/disk3s2   931Gi  922Gi  9.3Gi   100% 13356333          4281610946    0%   /Volumes/MacBack

███  RAM Usage
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                                9153.
Pages active:                           1574171.
Pages inactive:                         1565516.
Pages speculative:                         4948.
Pages throttled:                              0.
Pages wired down:                        494901.
Pages purgeable:                          16319.
"Translation faults":                1205143448.
Pages copy-on-write:                   94610989.
Pages zero filled:                    438277570.
Pages reactivated:                     39452701.
Pages purged:                           1082666.
File-backed pages:                       788531.
Anonymous pages:                        2356104.
Pages stored in compressor:             2503111.
Pages occupied by compressor:            541803.
Decompressions:                        17270952.
Compressions:                          25938006.
Pageins:                               24624213.
Pageouts:                                249515.
Swapins:                                2737920.
Swapouts:                               3495733.

███  RAM Usage per container
CONTAINER             CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
drupal8_varnish_1     0.68%               8.281MiB / 2.933GiB   0.28%               1.53kB / 0B         26.8MB / 475kB      219
drupal8_web_1         0.13%               5.258MiB / 2.933GiB   0.18%               1.53kB / 0B         4.68MB / 0B         82
drupal8_cli_1         0.03%               38.88MiB / 2.933GiB   1.29%               1.66kB / 3.66kB     113MB / 582kB       8
drupal8_db_1          0.07%               415.2MiB / 2.933GiB   13.83%              2.13kB / 0B         34.6MB / 397MB      21
drupal8_solr_1        0.11%               211.3MiB / 2.933GiB   7.04%               2.21kB / 0B         96.6MB / 81.9kB     37
drupal8_memcached_1   0.02%               1004KiB / 2.933GiB    0.03%               2.32kB / 0B         2.11MB / 0B         7
drupa7_web_1            0.10%               9.617MiB / 2.933GiB   0.32%               926kB / 2.24MB      6.66MB / 0B         109
drupa7_cli_1            0.02%               76.72MiB / 2.933GiB   2.55%               1.17GB / 567MB      1.16GB / 399MB      8
docksal-ssh-agent     0.00%               1.289MiB / 2.933GiB   0.04%               4.87kB / 0B         8.09MB / 0B         2
docksal-dns           0.35%               9.234MiB / 2.933GiB   0.31%               54.1kB / 74.9kB     16.6MB / 0B         17
docksal-vhost-proxy   0.32%               16.53MiB / 2.933GiB   0.55%               2.31MB / 2.32MB     41.4MB / 36.9kB     25
drupa7_db_1             0.07%               1.177GiB / 2.933GiB   40.13%              563MB / 1.05GB      277MB / 14.8GB      27
drupa7_bg-sync_1        0.00%               65.04MiB / 2.933GiB   2.17%               2.82kB / 0B         19.9MB / 12.8MB     4

  ```

</details>

**Comment by lmakarov** [comment]
@shelane `gethostbyname failure` error is some sort of DNS resolution issue. Are you able to `nslookup google.com` or `ping google.com` from `cli`?

Running `openssl s_client -connect google.com:443` produces the same results for me when run on my Mac and inside a `cli` container (both look good).

I noticed you are using `lpeabody/blt-cli` image for cli. Please try with an official one, e.g. `docksal/cli:2.1-php7.1`

**Comment by shelane** [comment]
I am trying to recreate what I did using the docksal/drupal8 example site. Replacing only the actual server name, this is what I did:
```
cd /tmp
curl -k https://certserver.mydomain.com/cspca.cer -o cspca.pem
openssl x509 -text -inform PEM -in cspca.pem >> cert.pem
```
When I tried to copy the cert.pem to `/etc/ssl/certs`, I get a permission denied error. This is running in `fin bash`. I was going to try to run as root, but I couldn't find the password for that anywhere in the documentation.

BTW, the lpeabody/blt-cli is an extension of the docksal image. It was mentioned in this issue: https://github.com/docksal/docksal/issues/428
